File: dev-repo/opportunityplus-260506_dev-deploy/UNOPS.PAO.ClientApp/src/app/features/partnerships/partners/models/organization-unit-relationship.model.ts
import { OrganizationHierarchyModel } from '@core/models/organization-hierarchy.model';

export interface OrganizationUnitRelationshipModel {
  organizationHierarchyId: number;
  organizationHierarchy?: OrganizationHierarchyModel;
  entityId: number;
  entityType: string;
} 

File: dev-repo/opportunityplus-260506_dev-deploy/UNOPS.PAO.Models/OrganizationUnits/OrganizationUnitRelationshipModel.cs
using System.Text.Json.Serialization;
using UNOPS.PAO.Domain.Entities;

namespace UNOPS.PAO.Models.OrganizationUnits;

public class OrganizationUnitRelationshipModel
{
    public int OrganizationHierarchyId { get; set; }
    public OrganizationHierarchyModel? OrganizationHierarchy { get; set; }
    public int EntityId { get; set; }
    public string EntityType { get; set; }
    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingDefault)]
    public EntityStatus Status { get; set; }
    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingDefault)]
    public bool IsDeleted { get; set; }
} 